sdk/data/azcosmos/cosmos_client.go

```diff
@@ -249,6 +276,14 @@ func (c *Client) NewQueryDatabasesPager(query string, o *QueryDatabasesOptions)
 			return page.ContinuationToken != nil
 		},
 		Fetcher: func(ctx context.Context, page *QueryDatabasesResponse) (QueryDatabasesResponse, error) {
+			var err error
+			// Move the span to the pager once https://github.com/Azure/azure-sdk-for-go/issues/23294 is fixed
+			spanName, err := getSpanNameForClient(c.accountEndpointUrl(), operationTypeQuery, resourceTypeDatabase, c.accountEndpointUrl().Hostname())
+			if err != nil {
+				return QueryDatabasesResponse{}, err
+			}
+			ctx, endSpan := azruntime.StartSpan(ctx, spanName.name, c.internal.Tracer(), &spanName.options)
+			defer func() { endSpan(err) }()
```

Comment:
we also need to report a few more attributes based on the response:
- db.cosmosdb.request_charge
- db.cosmosdb.status_code
- `db.cosmosdb.sub_status_code`

Reply:
These attributes only are available on the response (if succeeded) but Go SDK atm does not provide a way to add Attributes after span creation. They are already adding some common properties (like status code).

https://github.com/Azure/azure-sdk-for-go/blob/e8defc77367f50a8e61ba3cf9fbe221d81d4f39d/sdk/azcore/runtime/policy_http_trace.go#L23-L32

Reply:
> Go SDK atm does not provide a way to add Attributes after span creation

@jhendrixMSFT is there some trick? we need to be able to add attributes about the response for non-HTTP cases. I can create an issue and we can follow up  on the Cosmos SDK once it's possible.

Reply:
Created https://github.com/Azure/azure-sdk-for-go/issues/23308

Reply:
Hi! Would something like this work to set the response attributes?

```go
span := c.internal.Tracer().SpanFromContext(ctx)
span.SetAttributes([]tracing.Attribute{
	{Key: "db.cosmosdb.request_charge", Value: res.RequestCharge
})
```

Reply:
I don't know what's the expected contract for SpanFromContext if something creates children spans (or the current span has a parent?) If the answer to the linked issue is the above, we can certainly add it.

Reply:
If I'm reading the code correctly `azruntime.StartSpan` returns a new context in which the span is the only possible span for that specific context, so I believe that should work.

I did just test this out, and for some reason the attributes passed into `runtime.StartSpan` `spanName.options` aren't being displayed for us. I'm still trying to figure out why.

Reply:
>I did just test this PR out, and for some reason the attributes passed into runtime.StartSpan via spanName.options aren't being displayed for us. I'm still trying to figure out why.

Nevermind, this is working most of the time. It's just the span started by the `NewQueryItemsPager` `Fetcher` function that's not working for me. I see the `Pager[QueryItemsResponse].NextPage` span and the underlying HTTP POST span, but not the `query_items` span

Reply:
Yeah, the query's are still not working as expected (see the Issue I linked in that part in the code)

Reply:
Once that is fixed, then our span will appear (or we'll have to move the Span name and properties to a new input in the Pager)

Reply:
Yeah, I see that "by policy" nested spans are rejected in the tracer code, that makes sense. Thanks!

https://github.com/Azure/azure-sdk-for-go/blob/b31a752c22d8046d0e5d28ef4df05d111f2a6cf1/sdk/azcore/runtime/policy_http_trace.go#L120-L125

Reply:
@kjg for reference, this is the feature request we have https://github.com/Azure/azure-sdk-for-go/issues/23294

Reply:
Looks great, thanks for all of this, it is already quite valuable for us!